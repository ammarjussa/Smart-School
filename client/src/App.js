import React, { Component } from 'react';
import {Redirect, Switch, Route,BrowserRouter} from 'react-router-dom'
import Login from './Components/Login'
import AdminDashboard from './Components/Admin/AdminPanel'
import FacultyDashboard from './Components/Faculty/FacultyPanel'
import {ProtectedRoute} from './ProtectedRoute'
import NotFound from './Components/NotFound'
import './App.css';


class App extends Component {
  constructor(props) {
    super(props);
    this.state = {}
    this.connecToServer = this.connecToServer.bind(this);
  }
  connecToServer() {
    fetch('/');
  }

  componentDidMount() {
    this.connecToServer();
  }

  render() {
    return (
      <div>
      <BrowserRouter>
        <Switch>
          <Route exact path="/" component={Login}/>
          <ProtectedRoute path="/admin" component={AdminDashboard}/>
          <ProtectedRoute path="/faculty" component={FacultyDashboard}/>
          <Route path = "*" component={NotFound} />
        </Switch>
      
      </BrowserRouter>
      </div>
    );
  }
}

export default App
export {App}