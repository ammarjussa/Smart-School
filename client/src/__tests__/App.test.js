import {App} from '../App';
import {shallow} from 'enzyme'; 
import React from 'react'; 
import toJson from 'enzyme-to-json'; 

const component = shallow(<App />);

describe('Login Component', () => {
    it('renders without crashing ', () => {
        const props = {}
        const wrapper = shallow(<App {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
    
    it('Login provides login options for admin and faculty', () => {
        const wrapper = component.find('ProtectedRoute');
        expect(wrapper.length).toBe(2);
        // console.log(component.debug())

    })
})