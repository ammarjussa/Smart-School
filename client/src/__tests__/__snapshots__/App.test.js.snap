// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Component renders without crashing  1`] = `
<div>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <ProtectedRoute
        component={[Function]}
        path="/admin"
      />
      <ProtectedRoute
        component={[Function]}
        path="/faculty"
      />
      <Route
        component={[Function]}
        path="*"
      />
    </Switch>
  </BrowserRouter>
</div>
`;
